### PR TITLE
GitHub Action to lint Python code with ruff

### PR DESCRIPTION
**Pull request by cclauss**
## Proposed changes

Related to #8867

Provide contributors immediate feedback on [Python linting](https://beta.ruff.rs/) errors in pull requests with helpful GitHub Annotations that describe the problem and its location. This Action uses minimal steps to run in under 10 seconds to provide contributors with rapid yet intuitive feedback.
## Checklist

- [x] Lint and unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [ ] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

### CODE REVIEW

**Comment by nijel** [comment]
Thanks. I'm changing this to draft until it is ready. 

Moreover, I've created an issue for migration to ruff, see https://github.com/WeblateOrg/weblate/issues/8867.

File: .github/workflows/ruff.yml [hunk 1]
```diff
@@ -17,4 +17,4 @@ jobs:
     steps:
     - uses: actions/checkout@v3
     - run: pip install --user ruff
-    - run: ruff --format=github --ignore=E501,F405 .
+    - run: ruff --format=github --ignore=E501,E741,F401,F405 .
```
Comment (nijel) [inline .github/workflows/ruff.yml:20]: Right now, we ignore some of these in some files only, see: https://github.com/WeblateOrg/weblate/blob/d1c255a73f20131578918bd3546b339271d281d6/setup.cfg#L21-L40

Comment (cclauss) [reply .github/workflows/ruff.yml:20]: Cool. You can use https://pypi.org/project/ini2toml to migrate that all to `pyproject.toml` to get rid of `setup.py`.

Comment (nijel) [reply .github/workflows/ruff.yml:20]: I've just fixed these two in 1739c56206e5780c813e40fb7ef45dca1ebed529 and 919adfe8e2f4ff3f4a01dab40e3dd0430f5efe48:

```suggestion
    - run: ruff --format=github --ignore=E501,F405 .
```

**Pull request closed without merging**